tests/ci/worker/init_runner.sh

```diff
@@ -267,15 +283,21 @@ while true; do
         if ! is_job_assigned; then
             RUNNER_AGE=$(( $(date +%s) - $(stat -c +%Y /proc/"$runner_pid" 2>/dev/null || date +%s) ))
             echo "The runner is launched $RUNNER_AGE seconds ago and still has hot received the job"
-            if (( 60 < RUNNER_AGE )); then
+            if [ -z "$LABELS_DELETED" ] && (( 60 < RUNNER_AGE && RUNNER_AGE <= 120 )); then
+                # delete the tags, attempt to prevent another job assigned
+                curl -L -X DELETE -H "Authorization: Bearer $(get_runner_token)" \
```

Comment:
@Felixoid I wonder how this technique works out, can you share some info afterwards?

Reply:
You'll need PAT for that, you might be using the Runner Registration Token instead.

Reply:
the script become more and more complex..

Reply:
oh, it's actually much simpler than I thought!

```
ubuntu@ip-172-31-85-130:~/actions-runner$ ./config.sh remove --token "$(get_runner_token)"

# Runner removal

Failed: Removing runner from the server
Runner "i-0783da29f653d6fea" is still running a job"
ubuntu@ip-172-31-85-130:~/actions-runner$ echo $?
1
```

We already have basically ALL we need to prevent unattended shutting down. Just execute this removal in advance, before instance termination. If it fails, then the job is assigned. That's it